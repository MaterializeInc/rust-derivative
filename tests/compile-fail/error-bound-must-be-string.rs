#[cfg(feature = "use_core")]
extern crate core;

#[macro_use]
extern crate derivative;

#[derive(Derivative)]
#[derivative(Clone)]
struct Err1 {
    ok: u8,
    #[derivative(Clone(bound=42))]
    err1: u8,
}

#[derive(Derivative)]
#[derivative(Debug)]
struct Err2 {
    ok: u8,
    #[derivative(Debug(bound=42))]
    err2: u8,
}

#[derive(Derivative)]
#[derivative(PartialEq)]
struct Err3 {
    ok: u8,
    #[derivative(PartialEq(bound=42))]
    err3: u8,
}

#[derive(Derivative)]
#[derivative(Default)]
struct Err4 {
    ok: u8,
    #[derivative(Default(bound=42))]
    err4: u8,
}

#[derive(Derivative)]
#[derivative(Hash)]
struct Err5 {
    ok: u8,
    #[derivative(PartialEq(bound=42))]
    err5: u8,
}

#[derive(Derivative)]
#[derivative(Hash(bound=42))]
struct Err6 {
    ok: u8,
    #[derivative(PartialEq(bound=42))]
    err6: u8,
}

fn main() {}