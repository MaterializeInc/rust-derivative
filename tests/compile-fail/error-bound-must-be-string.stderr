error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:11:30
   |
11 |     #[derivative(Clone(bound=42))]
   |                              ^^

error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:19:30
   |
19 |     #[derivative(Debug(bound=42))]
   |                              ^^

error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:27:34
   |
27 |     #[derivative(PartialEq(bound=42))]
   |                                  ^^

error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:35:32
   |
35 |     #[derivative(Default(bound=42))]
   |                                ^^

error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:43:34
   |
43 |     #[derivative(PartialEq(bound=42))]
   |                                  ^^

error: expected derivative bound attribute to be a string: `bound = "..."`
  --> $DIR/error-bound-must-be-string.rs:48:25
   |
48 | #[derivative(Hash(bound=42))]
   |                         ^^
